src/distribute_kv_cache/rpc/message.rs

```diff
@@ -0,0 +1,1985 @@
+use std::{fmt, mem};
+
+use async_trait::async_trait;
+use bytes::{Buf, BufMut, BytesMut};
+use clippy_utilities::{Cast, OverflowArithmetic};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use tracing::{debug, error, warn};
+
+use super::{
+    error::RpcError,
+    packet::{ActualSize, Decode, Encode, Packet},
+    utils::{get_u64_from_buf, u64_to_usize},
+};
+use crate::async_fuse::util::usize_to_u64;
+
+/// Zero copy conversion between num key and u8 buffers.
+#[allow(clippy::mem_forget)]
+#[allow(clippy::as_conversions)]
+#[allow(dead_code)]
+fn num_to_u8_buffer<K>(input: Vec<K>) -> Vec<u8>
+where
+    K: num::Num,
+{
+    let len = input.len() * mem::size_of::<K>();
+    let ptr = input.as_ptr();
+    let capacity = input.capacity();
+
+    std::mem::forget(input);
+
+    unsafe { Vec::from_raw_parts(ptr as *mut u8, len, capacity) }
+}
+
+/// Zero copy conversion between u8 and num key buffers.
+#[allow(clippy::mem_forget)]
+#[allow(clippy::as_conversions)]
+#[allow(clippy::integer_division)]
+fn u8_to_num_buffer<K>(input: Vec<u8>) -> Vec<K>
+where
+    K: num::Num,
+{
+    assert_eq!(
+        input.len() % mem::size_of::<K>(),
+        0,
+        "Buffer length must be a multiple of key size"
+    );
+    let len = input.len() / mem::size_of::<K>();
+    let ptr = input.as_ptr();
+    let capacity = input.capacity() / mem::size_of::<K>();
+
+    std::mem::forget(input);
+
+    unsafe { Vec::from_raw_parts(ptr as *mut K, len, capacity) }
+}
+
+/// Zero copy conversion between u32 and u8 buffers.
+#[allow(dead_code)]
+#[allow(clippy::mem_forget)]
+#[allow(clippy::as_conversions)]
+#[allow(clippy::integer_division)]
+fn u32_to_u8_buffer(input: Vec<u32>) -> Vec<u8> {
+    let len = input.len() * 4;
+    let ptr = input.as_ptr();
+    let capacity = input.capacity() * 4;
+
+    std::mem::forget(input);
+
+    unsafe { Vec::from_raw_parts(ptr as *mut u8, len, capacity) }
+}
+
+/// Zero copy conversion between u8 and u32 buffers.
+#[allow(dead_code)]
+#[allow(clippy::mem_forget)]
+#[allow(clippy::cast_ptr_alignment)]
+#[allow(clippy::as_conversions)]
+#[allow(clippy::integer_division)]
+fn u8_to_u32_buffer(input: Vec<u8>) -> Vec<u32> {
```

Comment:
I don't think these two functions are necessary for that `num_to_u8_buffer` is enough.